scripts/lint-style.lean

```diff
@@ -104,6 +105,23 @@ def allScriptsDocumented : IO Bool := do
       {String.intercalate "," undocumented.toList}"
   return undocumented.size == 0
 
+/-- Verifies that all modules in `modules` are named in `UpperCamelCase`
+(except for explicitly discussed exceptions, which are hard-coded here). -/
+def checkModulesUpperCamelCase (modules : Array Lean.Name) : IO Bool := do
+  -- Exceptions to this list should be discussed on zulip!
+  let exceptions := [
+    `Mathlib.Analysis.CStarAlgebra.lpSpace,
+    `Mathlib.Analysis.InnerProductSpace.l2Space,
+    `Mathlib.Analysis.Normed.Lp.lpSpace
+  ]
+  -- We allow only names in UpperCamelCase, possibly with a trailing underscore.
+  let badNames := modules.filter fun name ↦
+    let upperCamelName := Lake.toUpperCamelCase name
+    !exceptions.contains name &&
+    (upperCamelName != name && s!"{upperCamelName}_" != name.toString)
+  for bad in badNames do
+    IO.eprintln s!"error: module name {bad} is not in 'UpperCamelCase': it should be {Lake.toUpperCamelCase bad} instead"
```

Comment:
```suggestion
    IO.eprintln s!"error: module name '{bad}' is not in UpperCamelCase: it should be '{Lake.toUpperCamelCase bad}' instead"
```
I have a slight preference for adding the quotes to the module names.

Reply:
Also, this function allows a single trailing `_` in the initial filter, but suggests stripping out the trailing `_`, in the error report.  Is that intentional?

Reply:
Not, it isn't. Good catch! I should (and will) add tests :-)

Reply:
Fixed and added a test (also for this behaviour). I'm actually not certain any more which of these is best, but the current solution seems good enough to me.